activerecord/test/cases/finder_test.rb

```diff
@@ -825,6 +825,12 @@ def test_second_to_last
     # test with limit
     assert_nil Topic.limit(1).second
     assert_nil Topic.limit(1).second_to_last
+
+    # test with order
+    c = Topic.connection
+    assert_sql(/ORDER BY #{Regexp.escape(c.quote_table_name("topics.updated_at"))} DESC LIMIT/i) do
+      Topic.order(:updated_at).second_to_last
+    end
```

Comment:
`test_second_to_last` and `test_third_to_last` seem to be concerned with the correctness of those methods, rather than implementation details.  What about moving these assertions to a separate test like `test_nth_to_last_with_order_uses_limit` (and only testing a single method, e.g. `second_to_last`)?

Reply:
👍 I also added a test for implicit ordering (it was missing in the mentioned commit in the description).